File: test/timing/kalman/kalman_speed.m
function [] = kalman_speed(node)
% Add paths
path = '/data/dsge_data_dir/proc/dsge/pearl/cleanCode990/';
dirs = {[path, 'data/']; [path, 'dsgesolv/']; [path, 'estimation/']; [path, 'forecast/']; ...
        [path, 'initialization/']; [path, 'kalman/']; [path, 'toolbox/']; [path, 'plotting/']};
addpath(dirs{:});

% Load variables
load('kalman.mat');

% Call Kalman filter
iterations = 1000;
tic;
for i = 1:iterations;
  [L, zend, Pend] = kalcvf2NaN(data, lead, a, F, b, H, variance, z0, vz0);
end;
time_elapsed = toc;

disp(['node: ', node, ' seconds: ',  num2str(time_elapsed)]);
exit;

end

File: test/timing/kalman/kalman_speed.jl
# Add paths
using DSGE, MATLAB
path = dirname(@__FILE__)

# Load variables
mf = MatFile("$path/kalman.mat")
data = get_variable(mf, "data")
lead = 1
a = reshape(get_variable(mf, "a"), 72, 1)
F = get_variable(mf, "F")
b = reshape(get_variable(mf, "b"), 12, 1)
H = get_variable(mf, "H")
variance = get_variable(mf, "variance")
z0 = reshape(get_variable(mf, "z0"), 72, 1)
vz0 = get_variable(mf, "vz0")
close(mf)

# The first time tic and toc are called, the function gets compiled, so we should ignore the returned time
tic()
for i = 1:1
    L, zend, Pend = kalcvf2NaN(data, lead, a, F, b, H, variance, z0, vz0)
end
toq()

# Call Kalman filter
iterations = 1000
tic()
for i = 1:iterations
    L, zend, Pend = kalcvf2NaN(data, lead, a, F, b, H, variance, z0, vz0)
end
time_elapsed = toq()

for node in ARGS
    println("node: $node, seconds: $time_elapsed")
end
